lib/diagnostics_channel.js

```diff
@@ -22,11 +29,59 @@ const { triggerUncaughtException } = internalBinding('errors');
 
 const { WeakReference } = internalBinding('util');
 
+function decRef(channel) {
+  if (channels.get(channel.name).decRef() === 0) {
+    channels.delete(channel.name);
+  }
+}
+
+function incRef(channel) {
+  channels.get(channel.name).incRef();
+}
+
+function markActive(channel) {
+  // eslint-disable-next-line no-use-before-define
+  ObjectSetPrototypeOf(channel, ActiveChannel.prototype);
+  channel._subscribers = [];
+  channel._stores = new SafeMap();
+}
+
+function maybeMarkInactive(channel) {
+  // When there are no more active subscribers or bound, restore to fast prototype.
+  if (!channel._subscribers.length && !channel._stores.size) {
+    // eslint-disable-next-line no-use-before-define
+    ObjectSetPrototypeOf(channel, Channel.prototype);
+    channel._subscribers = undefined;
+    channel._stores = undefined;
+  }
+}
+
+function defaultTransform(data) {
+  return data;
+}
+
+function wrapStoreRun(store, data, next, transform = defaultTransform) {
```

Comment:
Just a thought, but instead of a transform that replaces the context given to the store, couldn't `next` be passed to the function instead, which could then wrap it with `store.run()` or with any other functionality? This would decouple this from having to know explicitly about `AsyncLocalStorage`.